jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/component/Dumpable.java

```diff
@@ -182,7 +184,10 @@ else if (o instanceof Map.Entry<?, ?> entry)
     static void dumpObjects(Appendable out, String indent, Object object, Object... extraChildren) throws IOException
     {
         dumpObject(out, object);
-        
+        if (DumpAppendable.hasVisited(out, object))
+            return;
+
+        DumpAppendable.visit(out, object);
```

Comment:
I would isolate the "visit" code in this way:

```suggestion

        if (DumpAppendable.hasVisited(out, object))
            return;
        DumpAppendable.visit(out, object);

```